src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php

```diff
@@ -51,8 +53,20 @@ public function updateModuleStatus(array $modules): array
             if (in_array($module->getName(), $modules)
                 && array_key_exists($module->getName(), $modules)
                 && $module->getStatus() !== $modules[$module->getName()]) {
-                $module->setStatus((bool) $modules[$module->getName()]);
+                $module->setStatus((bool)$modules[$module->getName()]);
                 $this->getEntityManager()->persist($module);
+                //If the module is affecting the widget, update the relevant permissions
+                if ($module->getName() === 'leave') {
+                    $this->updateDataGroupPermissionForWidgetModules(
+                        'dashboard_leave_widget',
+                        (bool)$modules[$module->getName()]
+                    );
+                } elseif ($module->getName() === 'time') {
+                    $this->updateDataGroupPermissionForWidgetModules(
+                        'dashboard_time_widget',
+                        (bool)$modules[$module->getName()]
+                    );
+                }
```

Comment:
better to move this to service layer or API layer

Reply:
Added as event dispatcher